lib/ansible/module_utils/net_tools/nios/api.py

```diff
@@ -265,17 +265,34 @@ def run(self, ib_obj_type, ib_spec):
                 else:
                     proposed_object[key] = self.module.params[key]
 
+        # If configure_by_dns is set to False, then delete the default dns set in the param else throw exception
+        if not proposed_object.get('configure_for_dns') and proposed_object.get('view') == 'default' and ib_obj_type\
```

Comment:
I'm... surprised sanity doesn't yell at you for this? I would rather have the line break anywhere except in the middle of one of the tests.

I think sanity has opinions on whether the `and` should be at the end or the beginning, but whichever is the right one, I'd prefer to see that.